### PR TITLE
move to workspaces

### PR DESCRIPTION
**Pull request by anonthing**

### CODE REVIEW

File: myst/src/utils/config.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ impl Config {
             aws_secret: config.get_str("aws_secret").unwrap().to_string(),
             temp_data_path: config.get_str("temp_data_path").unwrap().to_string(),
             download_frequency: config.get_int("download_frequency").unwrap() as u64,
-            plugin_path: config.get_str("plugin_path").unwrap(),
+            plugin_path: config.get_str("plugin_path").unwrap_or(String::from("/etc/myst/plugin/metrics-reporter")),
```
Comment (mailravi3390) [inline myst/src/utils/config.rs:106]: plugin modules may typically come from /var or /opt

File: Dockerfile [hunk 2]
```diff
@@ -14,9 +14,11 @@ FROM debian:buster-slim
 RUN apt-get update \
     && apt-get -y install clang
 RUN mkdir -p /etc/myst          \
+    && mkdir -p /etc/myst/plugin \
     && mkdir -p /var/log/myst   \
     && mkdir -p /var/myst/data  \
     && mkdir -p /var/myst/tmp
+COPY --from=build /dep-download/target/release/libnoop_metrics_reporter.so /etc/myst/plugin/metrics-reporter
 COPY --from=build /dep-download/target/release/server /usr/bin/myst-server
 COPY --from=build /dep-download/target/release/segment-gen /usr/bin/myst-segment-gen
```
Comment (mailravi3390) [inline Dockerfile:23]: Will the binary gen path remain same or will it be under myst.

File: myst/src/utils/config.rs (hunk 1 above)
Comment (anonthing) [reply myst/src/utils/config.rs:106]: how about having all myst related binaries and plugins in `/usr/share/myst` , conf in `/etc/myst`  and data and logs in `/var/myst`

File: myst/src/utils/config.rs [hunk 3]
```diff
@@ -103,8 +110,14 @@ impl Config {
             aws_secret: config.get_str("aws_secret").unwrap().to_string(),
             temp_data_path: config.get_str("temp_data_path").unwrap().to_string(),
             download_frequency: config.get_int("download_frequency").unwrap() as u64,
-            plugin_path: config.get_str("plugin_path").unwrap(),
+            plugin_path: config.get_str("plugin_path").unwrap_or(String::from("/usr/share/myst/plugins/metrics-reporter")),
             ssl_for_metrics: config.get_bool("ssl_for_metrics").unwrap(),
+            rollup_size: config.get_int("rollup_size").unwrap_or(7*24*60*60) as u32,
+            num_containers: config.get_int("num_containers").unwrap_or(5) as usize,
+            container_id: config.get_int("container_id").unwrap_or(1) as usize,
```
Comment (mailravi3390) [inline myst/src/utils/config.rs:117]: num_containers and container_id will need to come from a separate file as these are dynamically injected.
Dynamic injection in k8s works via volumes.
The file name can be /etc/myst/shard-config/shard-config.toml or anything similar. We will mount the  /etc/myst/shard-config dir from outside.

**Review by mailravi3390 — APPROVED** [review]
👍

**Pull request merged**